docs/streams/developer-guide/dsl-topology-naming.html

```diff
@@ -0,0 +1,290 @@
+		      <-- KSTREAM-MAPVALUES-0000000002
+		 </code>
+
+		 From this report, we can see what the different operators are, but what is the broader context here?  
+		 For example, consider <code>KSTREAM-FILTER-0000000001</code>, we can see it's a 
+		 filter operation meaning records are dropped that don't match the given predicate.  But what is 
+		 the meaning of the predicate?  Additionally, we can see the topic names of the source and sink nodes, 
+		 but what if the topics aren't named in a meaningful way?  Then we're left to guess the 
+		 business purpose behind these topics.
+
+		 You'll also notice the numbering here; the source node is suffixed with <code>0000000000</code> 
+		 indicating it's the first processor in the topology.  
+		 The filter is suffixed with <code>0000000001</code> indicating it's the second processor in 
+		 the topology.   In the 2.4 release of Kafka Streams, there are now overloaded methods for 
+		 both <code>KStream</code> and <code>KTable</code> accepting 
+		 a new parameter <code>Named</code>.  By using the <code>Named</code> class DSL users can 
+		 now provide meaningful names to the processors in their topology.  
+
+		 Now let's take a look at your topology with naming all the processors:
+		 <code>
+		     builder.&gt;String,String&lt;stream("input", Consumed.as("Customer_transactions_input_topic"))
+		           .filter((k,v) -> !v.equals("invalid_txn"), Named.as("filter_out_invalid_txns"))
+		           .mapValues((v) -> v.substring(0,5), Named.as("Map_values_to_first_6_characters"))
+		           .to("output", Produced.as("Mapped_transactions_output_topic"));
+		 </code>
+
+		 <code>
+		 Topologies:
+		   Sub-topology: 0
+		    Source: Customer_transactions_input_topic (topics: [input])
+		      --> filter_out_invalid_txns
+		    Processor: filter_out_invalid_txns (stores: [])
+		      --> Map_values_to_first_6_characters
+		      <-- Customer_transactions_input_topic
+		    Processor: Map_values_to_first_6_characters (stores: [])
+		      --> Mapped_transactions_output_topic
+		      <-- filter_out_invalid_txns
+		    Sink: Mapped_transactions_output_topic (topic: output)
+		      <-- Map_values_to_first_6_characters
+		 </code>
+
+		Now you can look at the topology description and can easily understand what role each processor 
+		plays in the topology.  But there's another reason for naming your processor nodes when you 
+		have stateful operators that remain between restarts of your Kafka Streams applications, 
+		state stores, changelog topics, and repartition topics.
+
+		 <h2>Changing Names</h2>
+
+		 We've seen that the generated names are numbered where they are built in the topology.  
+		 This means that if you add/remove or shift the order of operations, the position of the 
+		 processor shifts, hence shifting the name of the processor.  Since <strong>most</strong> processors exist 
+		 in memory only this name shifting on topology changes presents no issue.  But the name 
```

Comment:
I'm finding the comment about **most** processors existing in memory only a bit confusing -- maybe just use the term "stateless" instead? Also, I think it's a bit misleading to say that "because most are stateless, this shifting presents no issue" -- maybe rephrase to something like "Many processors are stateless and therefore exist in memory only, so this name shifting on topology change presents no issue to applications built entirely of such operators." ?